azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -498,13 +545,26 @@ func resourceArmKubernetesClusterRead(d *schema.ResourceData, meta interface{})
 		if err := d.Set("service_principal", servicePrincipal); err != nil {
 			return fmt.Errorf("Error setting `service_principal`: %+v", err)
 		}
-	}
 
-	kubeConfigRaw, kubeConfig := flattenAzureRmKubernetesClusterAccessProfile(&profile)
-	d.Set("kube_config_raw", kubeConfigRaw)
+		aadProfile := flattenAzureRmKubernetesClusterAadProfile(resp.ManagedClusterProperties.AadProfile)
+		if err := d.Set("aad_profile", aadProfile); err != nil {
+			return fmt.Errorf("Error setting `aad_profile`: %+v", err)
+		}
+	}
 
-	if err := d.Set("kube_config", kubeConfig); err != nil {
-		return fmt.Errorf("Error setting `kube_config`: %+v", err)
+	serverAppId, ok := d.GetOkExists("server_app_id")
```

Comment:
Can we refactor the code to avoid duplication here? The main logic for non-AAD and AAD are almost the same.

Reply:
Refactored to remove duplication where possible.